File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int[] ans = new int[nums.length];
        int j = 0; 
        
        int pcount = 0;
        for (int i = 0; i < nums.length; i++){
            if (nums[i] < pivot){
                ans[j] = nums[i];
                j++;
            }
            else if (nums[i] == pivot){
                pcount++;
            }
        }
        
        for (int i = 0; i < pcount; i++){
            ans[j] = pivot;
            j++;
        }
        
        for (int i = 0; i < nums.length; i++){
            if (nums[i] > pivot){
                ans[j] = nums[i];
                j++;  
            }
        }
        
        return ans;
    }
}

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        l = 0
        ans = deque([num for num in nums if num == pivot])
        for r in reversed(range(len(nums))):
            if nums[l] > pivot:
                ans.append(nums[l])
            l += 1
            if nums[r] < pivot:
                ans.appendleft(nums[r])
        return list(ans)